### ISSUE TITLE
Send space button to continue until the shell prompt appears!?

### ISSUE THREAD

**Issue by CFRR** [phpseclib/phpseclib#1083]
Hi,

I'm testing phpseclib to access a Check Point Firewall and run some command.
One of the commands is a "show configuration" and during the execution a "--More--" appears and any key need to be pressed to continue.

How is possible to send for example "space button" every time that "--More--" is displayed until the shell prompt appears.

Thanks.

**Comment by metaclassing** [comment]
I use PHPSECLIB extensively for network device monitoring and configuration, I have found the most effective approach is altering the terminal or pager length.

I dont have a checkpoint to test on, however a quick googling suggested the following two options based on OS/appliance:

firewall> set clieenv rows 90000 (or some other very large number)
or
clish -c "show configuration"

Hopefully one of these options works out for you, alternatively you COULD try to use phpseclib ssh2 and REGEX prompt matching with more complex patterns to detect different hostname vs more prompt and send the appropriate response.

**Comment by terrafrost** [comment]
You can only send stuff once a command has been initiated if you have a PTY. `$ssh->read()` / `$ssh->write()` use PTY's implicitly. `$ssh->exec()` only uses a PTY if you did `$ssh->enablePTY()` first.

Anyway, with a PTY what you can do is something like this:

```php
while ($output = $ssh->read('#--More--|\[prompt\]#', NET_SSH2_READ_REGEX)) {
    if (!preg_match('#--More--$#', $output)) {
        break;
    }
    $ssh->write(' ');
}
```
(the regular expressions may need to be tweaked)

**Comment by CFRR** [comment]
Hi,

@metaclassing, this would be a good approach, if the limit was not 100.

@terrafrost, thanks, can you help me a lite more, please!?
I test the code that you suggest, and I'm getting the following error:

`>PHP Warning:  preg_match(): No ending delimiter '#' found in .../Net/SSH2.php on line 2797`

The Code:

```
<?php
include('Net/SSH2.php');

$ssh = new Net_SSH2('*');
if (!$ssh->login('admin', '*')) {
    exit('Login Failed');
}

$ssh->enablePTY();

echo $ssh->read('#');
$ssh->write("clish\n");
echo $ssh->read('>');
$ssh->write("lock database override\n");
echo $ssh->read('>');
$ssh->write("set virtual-system 3\n");
echo $ssh->read('>');
$ssh->write("show configuration\n");

while ($output = $ssh->read('#--More--|\[>\]', NET_SSH2_READ_REGEX)) {
    if (!preg_match('#--More--$#', $output)) {
        break;
    }
    $ssh->write(' ');
}
?>
```

**Comment by metaclassing** [comment]
One some platforms setting the pager limit to 0 disables it or means unlimited. Is that an option?

**Comment by CFRR** [comment]
@metaclassing, you are right, the "set clieenv rows 0" will show all configuration without stop! 👍

**Comment by terrafrost** [comment]
From the code snippet you posted:
```
while ($output = $ssh->read('#--More--|\[>\]', NET_SSH2_READ_REGEX)) {
```

You should be doing `'#--More--|\[>\]'#`. The first parameter of the `read()` call is, internally, passed to `preg_match` as the first parameter and `preg_match` needs to have the same delimiter at the start and end (unless you're using pattern modifiers (eg. `#...#s` is valid)

**Comment by swartzlib7** [comment]
terminal length 0

Ref: http://etherealmind.com/ios-setting-terminal-window-length/
